### PR TITLE
Decimal precision fix

### PR DESCRIPTION
**Pull request by aguibert**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -48,6 +48,8 @@
 
   <properties>
     <stack.version>4.0.0-SNAPSHOT</stack.version>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
```
Comment (aguibert) [inline pom.xml:52]: CC @vietj to make sure you are OK with this change

Comment (BillyYccc) [reply pom.xml:52]: Why do we need this as it's already defined in https://github.com/vert-x3/vertx-parent/blob/master/pom.xml?

Comment (aguibert) [reply pom.xml:52]: I thought I ran into an incompatible class version change and this fixed it, but I just double checked and it looks like we were indeed compiling to 1.8 before, so I must have been mistaken -- I'll remove this

File: vertx-db2-client/src/main/java/io/vertx/db2client/impl/drda/DRDAQueryRequest.java [hunk 2]
```diff
@@ -1399,7 +1409,7 @@ private Hashtable computeProtocolTypesAndLengths(
                     lidAndLengths[i][0]--;
                 }
             }
-            return overrideMap;
+            return overrideMap; // @AGG this is never used
```
Comment (aguibert) [inline vertx-db2-client/src/main/java/io/vertx/db2client/impl/drda/DRDAQueryRequest.java:1412]: Intentionally leaving this comment here so I remember to clean it up once I do the global cleanup of the drda package

File: vertx-db2-client/src/main/java/io/vertx/db2client/impl/drda/NetSqlca.java [hunk 3]
```diff
@@ -123,25 +123,31 @@ public static void throwSqlError(NetSqlca sqlca) {
...
+	   	    		throw new DB2Exception("The database '" + sqlErrmc + "' provided was not found", sqlca.sqlCode_, sqlca.sqlState_);
 	   	    	else
-	   	    		throw new DB2Exception("The database provided was not found", SqlCode.DATABASE_NOT_FOUND, sqlca.sqlState_);
+	   	    		throw new DB2Exception("The database provided was not found", sqlca.sqlCode_, sqlca.sqlState_);
+	   	    case SqlCode.DATA_TYPE_INVALID_ATTR:
+	   	        if (sqlErrmc.length() > 0)
+	   	            throw new DB2Exception("The statement cannot be processed. The data type definition '" + sqlErrmc + 
+	   	                "' specifies an invalid attribute such as precision, lenght, or scale.", sqlca.sqlCode_, sqlca.sqlState_);
+	   	        else
+	   	            throw new DB2Exception("The statement cannot be processed. A data type definition specifies an invalid attribute such as precision, lenght, or scale.", sqlca.sqlCode_, sqlca.sqlState_);
```
Comment (gjwatts) [inline vertx-db2-client/src/main/java/io/vertx/db2client/impl/drda/NetSqlca.java:150]: Same typo (copy/paste) :-)

File: vertx-db2-client/src/main/java/io/vertx/db2client/impl/drda/DRDAQueryRequest.java (hunk 2 above)
Comment (gjwatts) [reply vertx-db2-client/src/main/java/io/vertx/db2client/impl/drda/DRDAQueryRequest.java:1412]: Thanks for letting me know

**Pull request merged**